tests/integration/test_charm.py

```diff
@@ -29,6 +29,40 @@
 DUMMY_NAME = "app"
 
 
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy_same_machine(ops_test: OpsTest, kafka_charm):
```

Comment:
Really nice to have an integration test for this!

Reply:
**praise** I agree!